### ISSUE TITLE
[FEAT] Mods are watching

### ISSUE THREAD

**Issue by tigattack** [rHomelab/LabBot-Cogs#110]
# Overview
Mods are watching functionality, à la Araquiel's `:eye:` trigger.   
Example: 
![image](https://user-images.githubusercontent.com/10629864/105487270-b6bc0000-5ca7-11eb-8350-ed4373b0e246.png)

Cog should build these images on the fly to avoid issues with hardcoding. Suggest using imgflip API or similar. If using imgflip, use template ID 224894675.

I'm not sure if this is best as an extension of the [autoreply](https://github.com/rHomelab/LabBot-Cogs/tree/master/autoreply) cog or a cog of its own. Due to this, I've not filled out the suggested commands below.

# Commands
```
[p]suggested commands and usages
```

# DoD
* A member of the moderation team can post 👁️ (`:eye:`) and the bot responds as demonstrated above.

# Stretch
* Frog image, again replicating Araquiel's functionality. The implementation of this was not-so-ideal on the back end though. Suggest this is triggered by a mod posting 🐸 (`:frog:`) or similar. Open to suggestions here!
![image](https://user-images.githubusercontent.com/10629864/105487620-511c4380-5ca8-11eb-9246-465239a94b85.png)

**Comment by reichbc** [comment]
I suggest adding Horst of Disapproval for extra emphasis on the disapproval... 
![horse-of-disapproval](https://user-images.githubusercontent.com/30938747/106657226-a251f000-6550-11eb-87ea-dbcd45a7692e.jpg)

**Comment by marvelman3284** [comment]
I'm starting work on this feature and instead of using the imgflip api what are your thoughs on using the imgure api. The main reason I want to use imgur apit over the imgflip api is the [offical python library](https://github.com/Imgur/imgurpython) that can just speedline development. lmk which api you want me to use

**Comment by tigattack** [comment]
If you can produce the same end result with the imgur API, then that's fine :+1: I've created a mirror of the template [here](https://i.tiga.tech/LUlo2/nEpITUlO33.png/raw) if that's any help.

The only thing I notice is that while it is official, the imgurpython project has been deprecated since late 2017, so it may be better to find an alternative if possible.

I'd welcome input from @issy or any other member of the team who has any suggestions regarding this.

**Comment by marvelman3284** [comment]
Ok thanks. I realized that it's probably easier to use the imgflip api to create custom memes (ie. (Mod name) is watching).

**Comment by tigattack** [comment]
OK. You can use imgflip template ID 224894675.

**Comment by tigattack** [comment]
Since the template's upper text box is in the correct position there's no need to specify anything more than the text, so this is very easy:

```python
>>> import requests
>>> body = {'username':'redacted','password':'redacted','template_id':'224894675','font':'arial','boxes[0][text]':'your mum'}
>>> req = requests.post('https://api.imgflip.com/caption_image',data = body)
>>> print(req.text)
{"success":true,"data":{"url":"https:\/\/i.imgflip.com\/5b5vyr.jpg","page_url":"https:\/\/imgflip.com\/i\/5b5vyr"}}
```
[Resulting image.](https://i.imgflip.com/5b5vs9.jpg)

I'm sure there's a better way to do it, but this works fine as a PoC.

**Comment by issy** [comment]
Thanks for the snippet, Tig. That's definitely different from what I was doing :D
IIRC my method wasn't retaining font styling, but that seems to be retaining them perfectly.

**Comment by issy** [comment]
Suggested command usage RFC
```
[p]modmeme <enable|disable> <watching|frog|horse>
```

**Comment by marvelman3284** [comment]
This all looks really good! The frog and horse won't need any captioning right?

**Comment by issy** [comment]
Nope, they won't need any captioning. I've created imgflip templates for both of them, if that makes life easier.
Frog: 321087939
Horse: 321088192

**Comment by marvelman3284** [comment]
Also do you guys want a hardcoded username and password or is there some sorta config file that I can edit?

**Comment by issy** [comment]
Pull the username/password from the API key store built into Red. [Documentation here](https://docs.discord.red/en/stable/framework_apikeys.html)

**Comment by tigattack** [comment]
I'd like to suggest another command:
```
[p]modmeme trigger <watching|frog|horse> :emote: # Trigger conflicts should not be allowed.
```
Also, a note regarding the enable|disable command: I think it should default to be global enable/disable if the second param is not specified.

**Comment by marvelman3284** [comment]
> I'd like to suggest another command:
> 
> ```
> [p]modmeme trigger <watching|frog|horse> :emote: # Trigger conflicts should not be allowed.
> ```
> 
> Also, a note regarding the enable|disable command: I think it should default to be global enable/disable if the second param is not specified.

Noted. By trigger conflicts I assume you mean the auto reply triggers?

**Comment by tigattack** [comment]
Another idea I've just had:

Nicknames would be best to use for the template text since they can differ from the username and members may know a mod by their nickname rather than username.  
However, Discord has a limit of 32 characters for nicknames which could make the template unreadable or look poorly formatted.

I suggest that by default it uses the nickname, but has the following options available:
```
[p]modmeme setname <@modUser> <name> # Default to current user if user param not specified.
[p]modmeme setnamelimit <int> # Default to 32 chars if not specified.
```

I'm aware I'm causing a lot of scope creep here, sorry about that. Clearly I didn't think this through very thoroughly!

Think of these suggestions as stretches rather than DoD's.

**Comment by marvelman3284** [comment]
It's all fine. It will probably be a bit before I can get a working cog as I've never worked with RED before😅

**Comment by issy** [comment]
No worries 👍 having this discussion helped us clear up a lot of the implementation details anyway
